e2e_tests/index.js

```diff
@@ -0,0 +1,148 @@
+const express = require("express");
+const fs = require("node:fs").promises;
+const { spawn } = require("node:child_process");
+const httpProxy = require("http-proxy");
+
+class Streamsync {
+  constructor() {
+    this.process = null;
+    this.initialized = false;
+    this.port = 7358;
+    this.busy = false;
+  }
+
+  async start() {
+    return new Promise((resolve, reject) => {
+      if (this.process !== null) {
+        this.process.kill();
+      }
+      const ss = spawn(
+        "streamsync",
+        ["edit", "./runtime", "--port", this.port]
+      );
+      this.process = ss;
+      const startupTimeout = setTimeout(() => {
+        // eslint-disable-next-line no-console
+        console.error("Streamsync startup timeout");
+        ss.kill();
+        reject();
+      }, 5000);
+
+      ss.stdout.on("data", (data) => {
+        // eslint-disable-next-line no-console
+        console.log(
+          `[${ss.pid}] stdout: ${Buffer.from(data, "utf-8").toString()}`,
+        );
+        if (data.includes("Builder is available at")) {
```

Comment:
Any ideas to make this less hacky in the future? Something like a health check, some signal I don't know about?

Probably useful for other purposes too.